### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by adamruzicka**
TODOs:
- [x] triggering a job
- [x] polling for updates
- [x] error handling
  - [x] handling hosts which Satellite does not know - ack playbook run, consider them failed
  - [x] on initial trigger - fails immediatelly
  - [x] on output poll - retries 5 times, then fails
- [x] output diffs
- [x] configuration loading
- [x] determine result on finish
- [x] playbook run idempotency (don't run same run multiple times)
- [x] certificate auth
- [ ] remove debug messages

### CODE REVIEW

File: receptor_satellite/worker.py [hunk 1]
```diff
@@ -1,8 +1,50 @@
...
 import aiohttp
-# import functools
+import asyncio
 import json
 
+# TODO: Get this from somewhere
+SATELLITE_HOST = 'localhost:3000'
+
+
+class SatelliteAPI:
```
Comment (jhjaggars) [inline receptor_satellite/worker.py]: None of the functions defined under this class are actually methods (they don't take an initial self parameter).  Do you really need this class?

File: receptor_satellite/worker.py [hunk 2]
```diff
@@ -18,16 +60,76 @@ def __aiter__(self):
...
+        await self.put(payload)
+
+
+class Config:
+    def __init__(self, text_updates = False, text_update_interval = 5000, text_update_full = True):
+        self.text_updates = text_updates
+        self.text_update_interval = text_update_interval
+        self.text_update_full = text_update_full
+
+
```
Comment (jhjaggars) [inline receptor_satellite/worker.py:15]: It's usually better to make factory functions like this:

```
@classmethod
def myfun(cls, raw):
    return cls(raw['thingy', ...)
```

Comment (jhjaggars) [reply receptor_satellite/worker.py:15]: Also, be careful with mutable default args (like {} and []) as they are created only once when the module is compiled.  I think in this instance it's fine though.

File: receptor_satellite/worker.py [hunk 3]
```diff
@@ -18,16 +60,76 @@ def __aiter__(self):
...
+    def from_raw(raw = {}):
+        return Config(raw['text_updates'], raw['text_update_interval'], raw['text_update_full'])
+
+
+class Host:
```
Comment (jhjaggars) [inline receptor_satellite/worker.py:21]: Often when you have a class with two methods, one of which is __init__, you probably don't need a class.

File: receptor_satellite/worker.py [hunk 4]
```diff
@@ -1,8 +1,50 @@
...
+        return json.loads(response['body'])
+
+
+async def request(method, url, extra_data, queue):
+    try:
+        async with aiohttp.ClientSession() as session:
+            async with session.request(method, url, **extra_data) as response:
+                response_payload = dict(status=response.status,
+                                        body=await response.text())
+        return response_payload
```
Comment (jhjaggars) [inline receptor_satellite/worker.py]: Might as well put this under the same block as the request

File: receptor_satellite/worker.py (hunk 1 above)
Comment (adamruzicka) [reply receptor_satellite/worker.py]: used the class just as a logical grouping

Comment (jhjaggars) [reply receptor_satellite/worker.py]: In that case, you should decorate the functions with `@staticmethod`

Comment (adamruzicka) [reply receptor_satellite/worker.py]: Moved to a separate file, no decorator should be needed there I guess

**Review by matburt — COMMENTED** [review]
I think this just needs to add `asyncio` to the setup.py install dependencies and then we should get this merged.

There are some other changes that are coming in receptore core that will require changes to this but this is a good interim step.

**Pull request merged**